### ISSUE TITLE
pcustom does not take into account structure padding bytes

### ISSUE THREAD

**Issue by Grazfather** [hugsy/gef#446]
If you define a structure like this:

```python
from ctypes import *

class Bla(Structure):
    _fields_ = [
           ("field_0", c_ulong),
           ("field_8", c_uint),
           ("field_10", c_ulong),
    ]
```

field_10 is at offset 10, but in `apply_structure_to_address`, we print an offset that is only the sum of the sizes of the fields, so it'll display as 8 + 4 = 0xc, since it doesn't 'see' the 4 bytes of padding.

**Comment by hugsy** [comment]
> field_10 is at offset 10

Did you mean offset 0x10 ? Because no, it isn't:

```py
$ ipython3
>>> from ctypes import *
>>> sizeof(c_ulong)
Out[2]: 8
>>> sizeof(c_uint)
Out[3]: 4
```

Structures are not always packed by the compiler (i.e. `__attribute__((packed))`). Artificially adding padding shouldn't be gef's job, but the person's reimplementing the structure (for example it's not a problem when you export from IDA). Alternatively (and more cleanly IMHO) you have the option to change the alignment with the [`_pack_`](https://docs.python.org/3/library/ctypes.html#ctypes.Structure._pack_) Structure class attribute:

```py
class Bar(Structure):
  _fields_ = [ ("field_0", c_ulong),  ("field_8", c_uint), ("field_10", c_ulong), ]
>>> sizeof(Bla)
Out[10]: 24
>>> class Bar(Structure):
  _fields_ = [ ("field_0", c_ulong),  ("field_8", c_uint), ("field_10", c_ulong), ]
  _pack_=1
>>> sizeof(Bar)
Out[11]: 20
```
`_pack_` takes a small int that you can use for alignment (in your example, you can use 4 instead of 1).

Also FWIW, `ctypes` also uses the `stdint` standard so it's better to use `c_uint64`, `c_uint32`.

**Comment by Grazfather** [comment]
I've covered all this. Look at the original bug report where I found the bug (he was defining his structure wrong) (#406) and how I fixed it (#447)
